src/Pages/ChangePassword.cshtml.cs

```diff
@@ -58,15 +59,30 @@ public async Task<IActionResult> OnGet()
             {
                 return RedirectToPage("/index");
             }
-            //currency options nuskaitymo simuliacija
-            // Should probably do some backup thing, if the api doesn't work, it currently throws an exception if it doesn't
-            var currencyList = await CurrencyConverter.GetList();
+
+            await SetCurrency();
+            return Page();
+        }
+
+        private async Task SetCurrency()
+        {
+            //Some alert could be displayed that failed to get currency list
+            var userId = int.Parse(User.FindFirst(ClaimTypes.Name).Value);
+            UserModel = await UserDatabase.GetUserAsync(userId);
             CurrencyOptions = new List<string>();
-            foreach (var currency in currencyList)
+            try
             {
-                CurrencyOptions.Add(currency.Code);
+                var currencyList = await CurrencyConverter.GetList();
+                foreach (var currency in currencyList)
+                {
+                    CurrencyOptions.Add(currency.Code);
+                }
+            }
+            catch (Exception)
+            {
+                CurrencyOptions.Add(UserModel.Currency);
+                FailedToGetCurrencyList = true;
```

Comment:
As I understand most of code parts like these were changed when caching was implemented, so maybe it is not worth reviewing this?

Reply:
Yeah, I've already updated them in caching, and now updated this code again in the new branch I'm working on